add a config command

dev-mlb opened the pull request.

fbruton reviewed (changes requested):
I ran the following command and received duplicates:
`./emissary config --place emissary.place.sample.ToLowerPlace --offline --flavor test`
Recommend displaying unique and sorted entries  .

fbruton commented on src/main/java/emissary/server/api/Configs.java:
```diff
@@ -0,0 +1,182 @@
+     */
+    protected static String validate(String config) {
+        String name = StringUtils.appendIfMissing(StringUtils.trim(config), CONFIG_FILE_ENDING);
+        Paths.get(name);
```
Can this line be removed?

fbruton commented on src/main/java/emissary/server/api/Configs.java:
```diff
@@ -0,0 +1,182 @@
+        return new ConfigsResponseEntity(detailed);
+    }
+
+    static Pattern valid = Pattern.compile("^[A-za-z0-9._-]{1,255}$");
```
Will this updated regex address the CodeQL warning? Updated `A-z` to `A-Z`
`static Pattern valid = Pattern.compile("^[A-Za-z0-9._-]{1,255}$");`

fbruton reviewed (changes requested):
Added a couple of suggestions to address the CodeQL warnings.

jpdahlke commented:
Some observations:
 - The `-h` and `-p` flags do not appear to be honored, changing them locally prints that it's using my override, but doesn't actually leverage them. 
 - It also seems that the help displays far more options than are valid.
 - If the command is provided no parameters, a null.cfg is used rather than providing a response with "these are the expected flags"
 - I'm not sure I understand offline mode. If I provide no flavor to a running emissary, I would expect it to return it's effective configuration. If I DO provide a flavor, I would expect it to return a config based on what I gave, offline or not. If I'm trying to use the command and there is no running emissary then I can see needing to tell the command "don't try to connect". Would it make sense to simplify the options and say, "if flavor is provided, run in offline mode"?

dev-mlb commented on src/main/java/emissary/place/ServiceProviderPlace.java:
```diff
@@ -45,6 +46,18 @@
+    public static final String SERVICE_COST = "SERVICE_COST";
+    public static final String SERVICE_QUALITY = "SERVICE_QUALITY";
+    public static final String SERVICE_PROXY = "SERVICE_PROXY";
+    public static final String SERVICE_KEY = "SERVICE_KEY";
```
Yep, I can move these now that #367 was merged.

drivenflywheel reviewed (approved):
LGTM

dev-mlb commented:
> * The `-h` and `-p` flags do not appear to be honored, changing them locally prints that it's using my override, but doesn't actually leverage them.

If used in offline mode, h and p are used as startup params to generate the configs. If the configs have a call out to host and/or port, it'll use whatever you feed it. Otherwise, it is mainly needed for connecting to a specific node.

> * It also seems that the help displays far more options than are valid.

That is an Emissary thing. Same thing happens for agents command. We'll need a framework change to fix it. I'm happy to look into it to see if there is a way to toggle or override params from parent classes.

> * If the command is provided no parameters, a null.cfg is used rather than providing a response with "these are the expected flags"

Fixed this - made `--place` required

> * I'm not sure I understand offline mode. If I provide no flavor to a running emissary, I would expect it to return it's effective configuration. If I DO provide a flavor, I would expect it to return a config based on what I gave, offline or not. If I'm trying to use the command and there is no running emissary then I can see needing to tell the command "don't try to connect". Would it make sense to simplify the options and say, "if flavor is provided, run in offline mode"?

I added some validation to handle this. The output now looks like:
```
Command threw an exception: [config, --place, SamplePlace, --flavor, GRAPE]
com.beust.jcommander.ParameterException: --flavor can only be specified in offline mode
```
